src/com/facebook/buck/apple/XctoolRunTestsStep.java

```diff
@@ -96,7 +97,7 @@
   private final Optional<String> logLevelEnvironmentVariable;
   private final Optional<String> logLevel;
   private final Optional<Long> timeoutInMs;
-  private final Optional<Path> snapshotReferenceImagePath;
+  private final Optional<Either<Path, String>> snapshotReferenceImagesPath;
```

Comment:
I think you can simplify this by just making it a String, and pass in the String version of the path you get from `getResolver().getAbsolutePath(this.snapshotReferenceImagesPath.get().getLeft())` when calling the constructor for this.

Its important that this is a sourcepath at AppleTest.java line 113, so that the rulekey reference for the cache uses a relative path. After you've called absolute path on it though you can just treat this as a string. No use in two code paths if the only thing you do with the `Path` is call `toString` at the end.